src/Runner.php

```diff
@@ -333,11 +333,11 @@ public function init()
 
         // Create this class so it is autoloaded and sets up a bunch
         // of PHP_CodeSniffer-specific token type constants.
-        $tokens = new Tokens();
+        new Tokens();
```

Comment:
there are situations where the constructor can have side effect like documented in the comment above this line.

It is definitely a codesmell and could be cleaned up by moving the constants to some autolad file.

Reply:
I agree the way the `Tokens` file is set up is a codesmell, but changing it will be a breaking change as this is wired in like this in a couple of places, including for our own tests + tests for external standards.

We also cannot set a file to autoload via Composer as there are other supported installation manners + if these constants would be autoloaded, it is likely to break other tooling which has their own polyfills for these tokens (like PHP Parser, PHPStan etc).

I need to have a good think about this, but I don't think this will be fixed any time soon.